lib/ansible/cli/galaxy.py

```diff
@@ -104,13 +104,23 @@ def __init__(self, args):
         self._raw_args = args
         self._implicit_role = False
 
-        # Inject role into sys.argv[1] as a backwards compatibility step
-        if len(args) > 1 and args[1] not in ['-h', '--help', '--version'] and 'role' not in args and 'collection' not in args:
-            # TODO: Should we add a warning here and eventually deprecate the implicit role subcommand choice
-            # Remove this in Ansible 2.13 when we also remove -v as an option on the root parser for ansible-galaxy.
-            idx = 2 if args[1].startswith('-v') else 1
-            args.insert(idx, 'role')
-            self._implicit_role = True
+        if len(args) > 1:
+            # Inject role into sys.argv[1] as a backwards compatibility step
+            if args[1] not in ['-h', '--help', '--version'] and 'role' not in args and 'collection' not in args:
+                # TODO: Should we add a warning here and eventually deprecate the implicit role subcommand choice
+                # Remove this in Ansible 2.13 when we also remove -v as an option on the root parser for ansible-galaxy.
+                idx = 2 if args[1].startswith('-v') else 1
+                args.insert(idx, 'role')
+                self._implicit_role = True
+            # since argparse doesn't allow hidden subparsers, handle dead login arg from raw args after "role" normalization
+            if args[1:3] == ['role', 'login']:
```

Comment:
Wouldn't `ansible-galaxy role -v login` be an issue here? It seems like we should just be checking if `login` was set, or maybe displaying the error in `self.execute_login`.

Reply:
No worse than it was before- it'll just show the usage dump in that case with an invalid ROLE_ACTION (as it did prior to this change), since the subparser seems to require that the subcommand be first. I just didn't want to create any future trip hazards by globally blacklisting `login` as an arg value without using the parser. I tried doing this as a hidden option first, but it doesn't seem that you can do hidden subparsers in argparse. :(

Reply:
What if we were to keep the code here the same but move the error to `execute_login`. That way we are still relying on argparse to resolve the args like it does today and we don't have to complicate this already complicated bit of code.